miriway.cpp

```diff
@@ -66,27 +67,47 @@ std::map<std::string, ShellCommands::CmdFunctor> const wm_command =
         { "exit", [](ShellCommands* sc, bool shift) { sc->exit(shift); } },
     };
 
+struct ShellComponentRunInfo
+{
+    time_t last_run_time = 0;
+    int num_runs_with_too_short_time_in_between = 0;
+    std::unique_ptr<miral::FdHandle> handle = nullptr;
+};
+
+auto constexpr min_time_seconds_allowed_between_runs = 3;
+auto constexpr time_seconds_wait_foreach_run = 3;
+const int max_too_short_reruns_in_a_row = 10;
```

Comment:
These are only meaningful within `shell_launch`, hence they don't need to be at namespace scope.

Just make them `static` constants in the function body